### PR TITLE
v1.0.0 and multi-outputs

### PR DESCRIPTION
**Pull request by martinRenou**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dhirschfeld** [comment]
You saw the solve fail?
```
Mamba failed to solve:
 - jupytercad_core
 - python >=3.8
 - jupytercad_app
 - jupytercad_lab

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested jupytercad_core
-   - nothing provides requested jupytercad_app
-   - nothing provides requested jupytercad_lab
```

**Comment by martinRenou** [comment]
@dhirschfeld thank you for commenting! 

I assumed this could be ignored because those packages are actually in the multi-outputs recipe, I probably assumed wrong? How could I tell it to build those packages first?

**Comment by dhirschfeld** [comment]
I'm not sure - it does seem very weird, especially with the solve error right at the start.

I was just curious, but unfortunately I don't have any great insights to share - it doesn't seem to make much sense!

**Comment by dhirschfeld** [comment]
Are they all python arch packages except `jupytercad`?
```
 ['jupytercad_core-1.0.0-py312ha770c72_0.conda', 'jupytercad_lab-1.0.0-py312ha770c72_0.conda', 'jupytercad_app-1.0.0-py312ha770c72_0.conda', 'jupytercad-1.0.0-pyhd8ed1ab_0.conda']
```

**Comment by martinRenou** [comment]
Yeah it's weird!

> Are they all python arch packages except jupytercad?

They should all be noarch. Let's see if I force noarch Python for all of them. I assumed noarch: python at the top was enough.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,69 +1,176 @@
...
+    sha256: bfc98439ad4474f6f2c6100df5f9bd03a0af858adfd79538b55dde40152b163b
+    folder: jupytercad_app
+  - url: https://pypi.io/packages/source/j/jupytercad/jupytercad-{{ version }}.tar.gz
+    sha256: 4938583bde5a304569970301a04af11c6221d5364543a38ffde4329307a45e66
+    folder: jupytercad
 
 build:
   noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
   number: 0
```
Comment (dhirschfeld) [inline recipe/meta.yaml:24]: I might try putting the `noarch: python` into each of the recipes. Just a wild guess though :/

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,69 +1,176 @@
...
-    - jupyter labextension list  # [unix]
-    - jupyter labextension list 1>labextensions 2>&1  # [unix]
-    - cat labextensions | grep "@jupytercad/jupytercad-extension.*OK"  # [unix]
+outputs:
+  - name: jupytercad_core
+    version: {{ version }}
+    build:
+      script: |
+        cd jupytercad_core
+        python -m pip install . -vv --no-deps --no-build-isolation
```
Comment (dhirschfeld) [inline recipe/meta.yaml:34]: Can also instrument the build script:
```suggestion
      script: |
        set -euxo pipefail
        which pythonn
        cd jupytercad_core
        python -m pip install . -vv --no-deps --no-build-isolation
```

**Comment by dhirschfeld** [comment]
That seemed to work better:
```
['jupytercad_core-1.0.0-pyhd8ed1ab_0.conda', 'jupytercad_lab-1.0.0-pyhd8ed1ab_0.conda', 'jupytercad_app-1.0.0-pyhd8ed1ab_0.conda', 'jupytercad-1.0.0-pyhd8ed1ab_0.conda']
```

...but wasn't the root cause. Hopefully the name-shadowing does the job!

**Comment by martinRenou** [comment]
Name shadowing was indeed the issue! Thanks for the noarch pointer too.

I can continue debugging this :D

**Comment by dhirschfeld** [comment]
It looks like it's trying to download `pip` which seems a bit odd:
```
Successfully installed jupytercad_core-1.0.0
1 location(s) to search for versions of pip:
* https://pypi.org/simple/pip/
Fetching project page and analyzing links: https://pypi.org/simple/pip/
Getting page https://pypi.org/simple/pip/
Found index url https://pypi.org/simple/
Looking up "[https://pypi.org/simple/pip/"](https://pypi.org/simple/pip/%22) in the cache
Request header has "max_age" as 0, cache bypassed
No cache entry available
Starting new HTTPS connection (1): pypi.org:443
[https://pypi.org:443](https://pypi.org/) "GET /simple/pip/ HTTP/1.1" 200 27344
Updating cache with response from "[https://pypi.org/simple/pip/"](https://pypi.org/simple/pip/%22)
etag object cached for 1209600 seconds
Caching due to etag
Fetched page https://pypi.org/simple/pip/ as application/vnd.pypi.simple.v1+json
  Found link https://files.pythonhosted.org/packages/3d/9d/1e313763bdfb6a48977b65829c6ce2a43eaae29ea2f907c8bbef024a7219/pip-0.2.tar.gz (from https://pypi.org/simple/pip/), version: 0.2
  Found link https://files.pythonhosted.org/packages/18/ad/c0fe6cdfe1643a19ef027c7168572dac6283b80a384ddf21b75b921877da/pip-0.2.1.tar.gz
<snip>
```

You could try also adding `--ignore-installed --no-index` to the install commands:
https://github.com/conda/conda-build/blob/061063f10c51162707c7c5253daa39b2752fcfcc/conda_build/build.py#L3169-L3189

**Comment by dhirschfeld** [comment]
> I can continue debugging this :D

Good luck!

**Comment by martinRenou** [comment]
Let's go! Thanks

**Pull request merged**